### PR TITLE
Adds library implementation for EIP-712

### PR DESCRIPTION
**Pull request by mattrltrent**
## Primary changes

Introduces the following core functionality aligned with the [EIP-712 specification](https://eips.ethereum.org/EIPS/eip-712) for structured data hashing and signing:

- Implements the domain separator by hashing the domain type (`name`, `version`, `chainId`, `verifyingContract`) according to EIP-712.
 - Supports `chainId` using `uint256` to comply with large chain ID specification.
 - Properly handles the hashing of each domain field.
 - Introduces type validation for all types used in the `Message` and `Types` fields, ensuring only valid data types (i.e., `uint256`, `bytes32`, `string[]`) are allowed.
 - Includes support for dynamic and fixed-size arrays using Regex.
 - Implements `SignTypedData` to generate a signed message using ECDSA (`secp256k1`) private keys.
 - Introduces `RecoverAddrOfSigner` to recover the address from a signed message and verify its authenticity.
 - Adds `IsValidSignature` to check if a signature matches a specific address, which helps detect any tampering with the data.
 - Associated helper functions.
 
## Notes

- Includes 7 unit tests.
- Several references are included in the code.

### CODE REVIEW

File: lib/dids/eth.go [hunk 1]
```diff
@@ -46,9 +49,18 @@ func (d EthDID) Identifier() string {
...
 }
 
-func (d EthDID) Verify(payload apitypes.TypedData, sig string) (bool, error) {
-	// internally use RecoverSigner func to recover the signer DID from the sig
-	recoveredDID, err := d.recoverSigner(payload, sig)
+func (d EthDID) Verify(block blocks.Block, sig string) (bool, error) {
+	// internally use recoverSigner func to recover the signer DID from the sig
+
+	// convert the block to a typed data struct
+	payload, err := ConvertToEIP712TypedData("vsc.network", block, "tx_container_v0", func(f float64) (*big.Int, error) {
```
Comment (Geo25rey) [inline lib/dids/eth.go:56]: Block needs to be decoded as DAG-CBOR before being converted to Typed Data

File: lib/dids/eth.go [hunk 2]
```diff
@@ -46,9 +49,18 @@ func (d EthDID) Identifier() string {
 	return string(d)[len(EthDIDPrefix):]
 }
 
-func (d EthDID) Verify(payload apitypes.TypedData, sig string) (bool, error) {
-	// internally use RecoverSigner func to recover the signer DID from the sig
-	recoveredDID, err := d.recoverSigner(payload, sig)
+func (d EthDID) Verify(block blocks.Block, sig string) (bool, error) {
```
Comment (Geo25rey) [inline lib/dids/eth.go:52]: Missing unit test

**Comment by vaultec81** [comment]
As of latest known information. The issues related to Ethereum compatibility are related to the legacy Eth implementation. The implementation above is believed to be correct as this time.
Next steps:
- Parity checks with frontend implementation (as in impl proper JS version using metamask to confirm parity)
- Clean up git conflicts

**Comment by Geo25rey** [comment]
Continued in #14

**Pull request closed without merging**